diff --git a/src/RD_probe_cc_version.c b/src/RD_probe_cc_version.c
index a3e21c8..433e4bd 100644
--- a/src/RD_probe_cc_version.c
+++ b/src/RD_probe_cc_version.c
@@ -22,12 +22,11 @@ static
 int cc_version_callback(BYTE txStatus, BYTE rxStatus,
     ZW_APPLICATION_TX_BUFFER *pCmd, WORD cmdLength, void* user)
 {
-  if(!pCmd) return 0;
   rd_ep_database_entry_t *ep = (rd_ep_database_entry_t*) user;
   if (!ep) return 0;
   rd_node_database_entry_t* n = ep->node;
   if (!n) return 0;
-  if(txStatus == TRANSMIT_COMPLETE_OK) {
+  if(txStatus == TRANSMIT_COMPLETE_OK && !pCmd) {
     if(pCmd->ZW_VersionCommandClassReportFrame.requestedCommandClass == v.requestedCommandClass) {
       if(pCmd->ZW_VersionCommandClassReportFrame.commandClassVersion >= 0x01) {
         rd_node_cc_version_set(n, v.requestedCommandClass, pCmd->ZW_VersionCommandClassReportFrame.commandClassVersion);
@@ -60,12 +59,11 @@ static
 int version_capabilities_callback(BYTE txStatus, BYTE rxStatus,
     ZW_APPLICATION_TX_BUFFER *pCmd, WORD cmdLength, void* user)
 {
-  if(!pCmd) return 0;
   rd_ep_database_entry_t *ep = (rd_ep_database_entry_t*) user;
   if (!ep) return 0;
   rd_node_database_entry_t* n = ep->node;;
   if (!n) return 0;
-  if(txStatus == TRANSMIT_COMPLETE_OK) {
+  if(txStatus == TRANSMIT_COMPLETE_OK && !pCmd) {
     n->node_version_cap_and_zwave_sw = ((ZW_VERSION_CAPABILITIES_REPORT_V3_FRAME*)pCmd)->properties;
     n->pcvs->state = PCV_SEND_VERSION_ZWS_GET;
     pcv_fsm_post_event(ep, PCV_EV_VERSION_CAP_REPORT_RECV);
